FreeRTOS/FreeRTOS-Plus-TCP
# No ARP verification during DHCP phase

Description
-----------
This PR addresses [issue 414](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/issues/414) raised by @RuStrannik.

In his projects he calls `FreeRTOS_IPInit()` with zero's for all of the IP-addresses and netmask. That worked fine, until PR #338 was submitted: "Send ARP packets for unknown IP addresses". That PR made sure that the IP-addresses of incoming packets are verified by looking it up using ARP.

With the current PR, the ARP verification will be only be done after DHCP has set a proper IP-address and netmask.

Test Steps
-----------
Set `ipconfigUSE_DHCP` to 1 in your FreeRTOSIPConfig.h
Initialise the IP-stack as show in the issue:
~~~c
    const uint8_t addr[4] = { 0, 0, 0, 0 };
    FreeRTOS_IPInit(addr, addr, addr, addr, ucMacAddress);
~~~
Now boot the device with or without the change in FreeRTOS_UDP_IP.c

---
Most developers **never noticed** the above problem because `FreeRTOS_IPInit()` is normally not called with a zero netmask.

Thank you @RuStrannik for taking the effort to create the issue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
